### DISCUSSION TITLE
Dispatch some generic events

### DISCUSSION THREAD

**Discussion by agentphoenix** [awcodes/scribble#33] [Ideas]
I know that Scribble dispatches an event specific to the settings modal that you're using, but it would be awesome if there was a generic event dispatched so I know when a block as been inserted or updated. (In my use case, I want to trigger the model to save when a block is inserted or updated for my page designer.)

In my app I've just extended the Scribble modal and I dispatch an event before the save happens, but I'd love to not have to extend like this.

```php
abstract class ScribbleModal extends BaseScribbleModal
{
    public function save(): void
    {
        $this->dispatch('saved-scribble-modal');

        parent::save();
    }
}
```

I'd also love to be able to do something based on when a block is removed. I've found the Svelte code that gets called, but I don't see anything where I would be able to hook in to that event in my Livewire component.

**Comment by awcodes** [comment]
Yea. This can get complicated pretty quickly, because you also have to check against the statePath in the event as well since there can be more than one editor on the page at a time all listening for the same event.

**Comment by awcodes** [comment]
All the event are dispatched to the window though. So you be able to listen for them anywhere.

**Reply by agentphoenix** [reply]
This all makes sense. I didn't think about the multiple editors in a form thing.

So if I wanted to take an action from my Livewire component each time that a tool was inserted or updated (in this case saving a model), is there a built-in way to do that?

**Reply by awcodes** [reply]
You'd have to listen for the 'handle-{block-identifier}' event then you can check against the context for if it is 'insert' or 'update'

**Reply by awcodes** [reply]
Unfortunately, there's not a great way to have specific events since each tool can be 5 different types of tools which each have their own events based on how they interact with the editor.

I am open to suggestions of a good way to clean that up though.
